src/EditorFeatures/CSharpTest/UseRecursivePatterns/UseRecursivePatternsTests.cs

```diff
@@ -0,0 +1,239 @@
+            {
+                Name: string.Empty, ContainingNamespace: null
+            }
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" &&
+            type.ContainingSymbol is var declContainer &&
+            declContainer.Kind == SymbolKind.Namespace [||]&&
+            declContainer.Name == ""System"";
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", ContainingSymbol:
+            {
+                Name: ""System"", Kind: SymbolKind.Namespace
+            }
+
+            declContainer
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test4()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" &&
+            type.ContainingSymbol is SomeType declContainer &&
+            declContainer.Kind == SymbolKind.Namespace [||]&&
+            declContainer.Name == ""System"";
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", ContainingSymbol: SomeType
+            {
+                Name: ""System"", Kind: SymbolKind.Namespace
+            }
+
+            declContainer
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test5()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" [||]&& type.IsStruct;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", IsStruct: true
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test6()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.Name == ""ValueTuple"" [||]&& !type.IsClass;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            Name: ""ValueTuple"", IsClass: false
+        };
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Test7()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M()
+    {
+        return type.ContainingSymbol is var declContainer &&
+            declContainer.Kind == SymbolKind.Namespace &&
+            declContainer.Name == ""System"" [||]&&
+            (declContainer.ContainingSymbol as NamespaceSymbol)?.IsGlobalNamespace == true;
+    }
+}",
+@"class C
+{
+    bool M()
+    {
+        return type is
+        {
+            ContainingSymbol:
+            {
+                ContainingSymbol: NamespaceSymbol
+                {
+                    IsGlobalNamespace: true
+                }
+
+                , Name: ""System"", Kind: SymbolKind.Namespace
+            }
+
+            declContainer
```

Comment:
?

Reply:
this is the pattern designation which comes from the var pattern. see comment above.